README.md

```diff
@@ -77,6 +78,19 @@ A function which implements the following interface:
 
 The `onChunk` handler will be invoked each time a chunk of data it returned by the server. This function will be invoked one or more times depending on the response.  The function is invoked with two arguments; the first is an optional error which will be null unless there was a parsing error thrown by the `chunkParser``.  The second argument is an optional parsedChunk value which is produced by the supplied `chunkParser` (see: `options.chunkParser`).
 
+#### onHeaders (optional)
+A function which implements the following interface:
+
+```js
+(headers, statusCode) => undefined
+```
+
+A function which will be invoked once when the browser has returned the headers of the response. This function is invoked with two arguments:
```

Comment:
please can you add a note on the temporal coupling here (ie: does this get invoked before the first chunk?)